hack/generator/pkg/astmodel/type.go

```diff
@@ -31,6 +31,19 @@ type Type interface {
 	Equals(t Type) bool
 }
 
+// Types is the set of all types being generated
+type Types map[TypeName]TypeDefinition
+
+// Add adds a type to the set, with safety check that it has not already been defined
+func (types Types) Add(def TypeDefinition) {
+	key := def.Name()
+	if _, ok := types[key]; ok {
+		panic(fmt.Sprintf("type already defined: %v", key))
```

Comment:
Do we prefer this over err return?

Reply:
it's not something that can be dealt with by the caller, it's a code-correctness issue